infra/docker-swarm/ansible/roles/testground-services/tasks/main.yaml

```diff
@@ -0,0 +1,53 @@
+- name: Create Testground overlay control network
+  docker_network:
+    name: control
+    driver: overlay
+    attachable: yes
+    enable_ipv6: no
+    ipam_config:
+      - subnet: 10.1.0.0/16
+
+- name: Pull ipfs/testground image
+  docker_image:
+    name: "ipfs/testground:latest"
+    source: pull
+    force_source: yes
+    state: present
+
+- name: Pull Redis image
+  docker_image:
+    name: "redis:latest"
+    source: pull
+    force_source: yes
+    state: present
+
+- name: Start Redis service
+  docker_swarm_service:
+    name: testground-redis
+    image: "redis:latest"
+    command: redis-server
+    args:
+      - --notify-keyspace-events $szxK
```

Comment:
No need to escape `$`? It's an unfortunate choice by Redis. It represents string commands: https://redis.io/topics/notifications#configuration

Reply:
This is something I will be reviewing next. Redis config doesn't seem correct right now, but this also depends on the service discovery we have coded in Testground.